wire-protoc-integration/src/main/java/com/squareup/wire/protocwire/Plugin.kt

```diff
@@ -0,0 +1,260 @@
+ *
+ * <pre>`
+ * @JvmStatic
+ * fun main(args: Array<String>) {
+ *   Plugin.run(Generator())
+ * }
+`</pre> *
+ * To get protoc to use the plugin, you'll have to create a launcher script:
+ * <pre>`
+ * #!/bin/sh
+ * cd `dirname $0`
+ * exec java -jar myplugin.jar
+`</pre> *
+ * On Windows, if it lives in the same folder as the plugin's JAR, it will
+ * probably look like:
+ * <pre>`
+ * @echo off
+ * java -jar "%~dp0\myplugin.jar"
+ * exit %errorlevel%
+`</pre> *
+ * You'll then have to do one of the following:
+ *
+ *  * Place the plugin binary somewhere in the `PATH` and give it the
+ * name `protoc-gen-NAME` (replacing `NAME` with the name of
+ * your plugin). If you then invoke protoc with the parameter
+ * `--NAME_out=OUT_DIR` (again, replace `NAME` with your
+ * plugin's name), protoc will invoke your plugin to generate the output,
+ * which will be placed in `OUT_DIR`.
+ *  * Place the plugin binary anywhere, with any name, and pass the
+ * `--plugin` parameter to protoc to direct it to your plugin like
+ * so:
+ * <pre>
+ * protoc --plugin=protoc-gen-NAME=path/to/myscript --NAME_out=OUT_DIR
+</pre> *
+ * On Windows, make sure to include the `.bat` suffix:
+ * <pre>
+ * protoc --plugin=protoc-gen-NAME=path/to/myscript.bat --NAME_out=OUT_DIR
+</pre> *
+ *
+ *
+ * @author t.broyer@ltgt.net Thomas Broyer
+ * <br></br>Based on the initial work of:
+ * @author kenton@google.com Kenton Varda
+ */
+object Plugin {
+    /**
+     * Runs the given code generator, reading the request from [System.in]
+     * and writing the response to [System.out].
+     *
+     * @see .run
+     */
+    @JvmOverloads
+    fun run(generator: CodeGenerator, environment: Environment = DefaultEnvironment()) {
+        var request: PluginProtos.CodeGeneratorRequest
+        val rawRequest: ByteString
+        try {
+            rawRequest = ByteString.readFrom(environment.inputStream())
+            request = PluginProtos.CodeGeneratorRequest.parseFrom(rawRequest)
+        } catch (e: IOException) {
+            throw PluginException("protoc sent unparseable request to plugin.", e)
+        }
+        var files = asDescriptors(request.protoFileList)
+        val reg = createExtensionRegistry(files.values)
+
+        // now we must *re-parse* the request, but this time we can properly parse any
+        // custom options therein
+        request = try {
+            PluginProtos.CodeGeneratorRequest.parseFrom(rawRequest, reg)
+        } catch (e: IOException) {
+            throw PluginException("protoc sent unparseable request to plugin.", e)
+        }
+        files = asDescriptors(request.protoFileList)
+        val output = CodedOutputStream.newInstance(
+            environment.outputStream()
+        )
+        try {
+            // go ahead and write response preamble
+            PluginProtos.CodeGeneratorResponse
+                .newBuilder() // add more here as more features are introduced and then supported
```

Comment:
Is it a TODO?, and let's keep comments in their own line, that helps readability

Reply:
Removed